### ISSUE TITLE
Question about entropy threshold definition in Figure 4

### ISSUE THREAD

**Issue by miachen0401** [WillDreamer/ARL-Arena#1]
Hi, thanks for open-sourcing this work — I really enjoyed the paper and the repo.

I have a question regarding the entropy threshold τ used in **Figure 4**.

From the visualization, it appears that the total share of categories with **E < τ** and **E > τ** stays exactly **50/50** throughout training. If that reading is correct, I am confused about how τ is defined.

My understanding is that during RL training, the overall policy entropy should generally decrease over time. So if τ were a fixed constant, I would expect the proportion of samples satisfying **E < τ** to gradually increase, rather than remain balanced at 50/50.

Could you clarify the following?

- How is the entropy threshold τ in Figure 4 defined?
- Is τ a fixed constant, or recomputed dynamically during training?
- If it is dynamic, is it based on a statistic of the current entropy distribution (for example, a median / percentile split)?

I might be misunderstanding the figure, so any clarification would be greatly appreciated.

Thanks again for the great work.

**Comment by WillDreamer** [comment]
Hi, thanks for your interest!

1. The entropy threshold \tau is defined as the mean value. This implies that we must first traverse all trajectories in the current step to compute the average entropy, and only then can we proceed with the plotting.
```
entropy_mean = 0
for idx in range(len(tensor_data)):
    log_probs = tensor_data[idx]['log_probs']
    reverse_idx = -1
    for i, val in enumerate(reversed(log_probs), 1):  # 1-based index from right
        if val != 0:
            reverse_idx = i
            break
    mask_idx = len(log_probs) - reverse_idx

    tensor_data[idx]['mask_idx'] = mask_idx
    entropy = np.array(tensor_data[idx].get('entropy', 0))[:mask_idx]

    entropy_mean += entropy.mean()
entropy_mean /= len(tensor_data)
```

2. You said right. During RL training, the overall policy entropy should generally decrease over time. However, it should be noted that the KL values vary significantly across different steps. To produce clearer visualizations, we apply a **softmax** normalization to the eight KL values within each step before plotting. As a result, the curves appear relatively balanced in the figure, although the underlying absolute values are in fact fluctuating. The plotting code is also available at: https://github.com/WillDreamer/ARL-Arena/blob/main/scripts/stream_ploy.ipynb for your reference.

Thanks again for your interest!
